pkg/render/logstorage.go

```diff
@@ -499,15 +495,32 @@ func (es elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
 			}},
 		}
 		initContainers = append(initContainers, initKeystore)
+		annotations = es.dexCfg.RequiredAnnotations()
 	}
 
 	podTemplate := corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: annotations,
+		},
 		Spec: corev1.PodSpec{
 			InitContainers:     initContainers,
 			Containers:         []corev1.Container{esContainer},
 			ImagePullSecrets:   getImagePullSecretReferenceList(es.pullSecrets),
 			NodeSelector:       es.logStorage.Spec.DataNodeSelector,
 			ServiceAccountName: "tigera-elasticsearch",
+			Volumes: []corev1.Volume{
```

Comment:
Could you  please add a RequiredVolumes() method to the dexConfig to get the volumes so they don't need to be created here.